### PR TITLE
MGDSTRM-9146 - Admin server does not return latest from all partitions

### PR DESCRIPTION
**Pull request by biswassri**

### CODE REVIEW

File: systemtests/src/test/java/org/bf2/admin/kafka/systemtest/plain/RecordEndpointTestIT.java [hunk 1]
```diff
@@ -229,23 +230,24 @@ void testConsumeRecordsByStartingOffset(int startingOffset, int expectedResults)
...
+        "20",
+        "30",
+        "40",
+        "200"
     })
     void testConsumeLatestRecords(int limit) {
         final String topicName = UUID.randomUUID().toString();
-        final int totalRecords = 10;
-        topicUtils.createTopics(List.of(topicName), 1, Status.CREATED); // single partition
+        final int totalRecords = 20;
```
Comment (MikeEdgar) [inline systemtests/src/test/java/org/bf2/admin/kafka/systemtest/plain/RecordEndpointTestIT.java:241]: Try using `100` here. That way the cases for `limit`s less than `200` will be selecting fewer than the total like the previous version.
```suggestion
        final int totalRecords = 100;
```

File: systemtests/src/test/java/org/bf2/admin/kafka/systemtest/plain/RecordEndpointTestIT.java [hunk 2]
```diff
@@ -229,23 +230,24 @@ void testConsumeRecordsByStartingOffset(int startingOffset, int expectedResults)
...
-        "3",
-        "4",
-        "20"
+        "10",
+        "20",
+        "30",
+        "40",
+        "200"
     })
     void testConsumeLatestRecords(int limit) {
```
Comment (MikeEdgar) [inline systemtests/src/test/java/org/bf2/admin/kafka/systemtest/plain/RecordEndpointTestIT.java:239]: Looks like the issue is with the fixed `partition` parameter in the fetch. Try removing `.queryParam("partition", 0)` further down in this test case.

**Review by MikeEdgar — CHANGES_REQUESTED** [review]

File: kafka-admin/src/main/java/org/bf2/admin/kafka/admin/RecordOperations.java [hunk 3]
```diff
@@ -137,6 +138,9 @@ public ConsumerRecords<byte[], byte[]> next() {
                     int pollSize = records.count();
                     emptyPoll = pollSize == 0;
                     recordsConsumed.addAndGet(pollSize);
+                    if (recordsConsumed == null) {
+                        throw new NoSuchElementException("No records were consumed");
+                    }
```
Comment (MikeEdgar) [inline kafka-admin/src/main/java/org/bf2/admin/kafka/admin/RecordOperations.java:143]: `recordsConsumed` will never be null (assigned once on line 125). I think the simplest thing for this exception would be to put the following as the first thing in this `next` method:

```java
if (!hasNext()) {
    throw new NoSuchElementException();
}
```

Comment (biswassri) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/RecordOperations.java:143]: Okaay! Makes sense. When I tried with `if ( emptyPoll ) { throw new NoSuchElementException(); }`  it was throwing an incorrect status code for the other system tests. So, I tried this one.  But it makes sense to have the `!hasNext` condition for the exception. 
Thanks Mike!

Comment (MikeEdgar) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/RecordOperations.java:143]: Sounds good, thanks. If the systemtests have issues with the change, let me know. They might be incorrect.

Comment (biswassri) [reply kafka-admin/src/main/java/org/bf2/admin/kafka/admin/RecordOperations.java]: That worked perfectly Mike! Thanks again.

**Review by MikeEdgar — APPROVED** [review]
LGTM :+1: Thanks @biswassri

**Review by rareddy — APPROVED** [review]
lgtm

I do not think I quite followed the lamda based final call that brings everything together :) just that I was never used StreamSupport class before

**Pull request merged**